src/store/traction/pacbio/runCreate/actions.js

```diff
@@ -1,4 +1,47 @@
 import { handleResponse } from '@/api/ResponseHelper'
+import { groupIncludedByResource } from '@/api/JsonApi'
+
+const findPools = async ({ commit, getters }, filter) => {
+  // when users search for nothing, prompt them to enter a barcode
+  if (filter['barcode'].trim() === '') {
+    return {
+      success: false,
+      errors: ['Please provide a plate barcode'],
+    }
+  }
+
+  const request = getters.poolRequest
+  const promise = request.get({
+    include: 'tube,libraries.tag,libraries.request',
+    fields: {
+      requests: 'sample_name',
+      tubes: 'barcode',
+      tags: 'group_id',
+      libraries: 'request,tag,run_suitability',
+    },
+    filter,
+  })
+  const response = await handleResponse(promise)
+
+  const { success, data: { data, included = [] } = {}, errors = [] } = response
+  // success is true with an empty list when no pools match the filter
+  if (success && data.length > 0) {
+    const { tubes, libraries, tags, requests } = groupIncludedByResource(included)
+
+    commit('setPools', data)
+    commit('setTubes', tubes)
+    commit('setLibraries', libraries)
+    commit('setTags', tags)
+    commit('setRequests', requests)
+    return { success, errors }
+  } else {
+    return { success: false, errors: [`Unable to find pool with barcode: ${filter['barcode']}`] }
+  }
+}
+
+const actions = { findPools }
+
+export { findPools }
```

Comment:
Exporting is a bit weird in this file. I think you can remove like 42 & 44 and move the find pools function below into `export default { ... } `